de.dlr.sc.virsat.uiengine.ui/src/de/dlr/sc/virsat/uiengine/ui/editor/snippets/general/UiSnippetRoleManagement.java

```diff
@@ -76,19 +89,53 @@ protected String getSectionDescription() {
 		return SECTION_DESCRIPTION_PREFIX + user;
 	}
 
+    /**
+     * Open a custom dialog to manage users for a discipline.
+     */
+	private void openCustomDialog() {
+	    Table table = tableViewer.getTable();
+	    int selectionIndex = table.getSelectionIndex();
+
+	    if (selectionIndex != -1) {
+	        Discipline selectedDiscipline = (Discipline) tableViewer.getElementAt(selectionIndex);
+
+	        // Extract the users for the selected discipline
+	        String[] existingUsernames = selectedDiscipline.getUsers().toArray(new String[0]);
+
+	        // Open the CustomDialog and pass the callback
+	        CustomDialog customDialog = new CustomDialog(Display.getCurrent().getActiveShell(), existingUsernames, this);
+	        customDialog.open();
+	    }
+	}
+
+
 	@Override
 	protected void createTableColumns(EditingDomain editingDomain) {
+		
 		// Column Discipline Name
 		TableViewerColumn columnName = createDefaultColumn(tableViewer, COLUMN_TEXT_DISCIPLINE);
 		columnName.setLabelProvider(getDefaultColumnLabelProvider(editingDomain, true, GeneralPackage.Literals.INAME__NAME));
 		columnName.setEditingSupport(new EStringCellEditingSupport(editingDomain, tableViewer, GeneralPackage.Literals.INAME__NAME));
 		
 		// Column of the user name
 		TableViewerColumn columnUser = createDefaultColumn(tableViewer, COLUMN_TEXT_USER);
-		columnUser.setLabelProvider(getDefaultColumnLabelProvider(editingDomain, false, RolesPackage.Literals.DISCIPLINE__USER));
-		columnUser.setEditingSupport(new EStringCellEditingSupport(editingDomain, tableViewer, RolesPackage.Literals.DISCIPLINE__USER));		
+		columnUser.setLabelProvider(getDefaultColumnLabelProvider(editingDomain, false, RolesPackage.Literals.DISCIPLINE__USERS));
+		columnUser.setEditingSupport(new EListStringCellEditingSupport(editingDomain, tableViewer, RolesPackage.Literals.DISCIPLINE__USERS));
+	    // Pack the column to adjust its width based on the content

```

Comment:
Looks like the identation is weird here?

Reply:
Yes sure. I did the mistake. but i remove this columnUser.setEditingSupport(new EStringCellEditingSupport(editingDomain, tableViewer, RolesPackage.Literals.DISCIPLINE__USER));.

Reply:
I was talking about the white spaces here...